fix: make code work again (#20525)
mdn/content: files/en-us/web/api/gamepad_api/using_the_gamepad_api/index.md

    scangamepads();
  }

>   for (const i in controllers) {
>     const controller = controllers[i];

Reviewer: Never use `for...in` with arrays and objects (as extra methods or properties can be added). While `forEach` doesn't work with objects, it is better to do this.
```suggestion
  controllers.entries().forEach((controller, i) => {
```

Author: As controllers is an object (map of gamepads by id) I used `Object.entries`

      const pct = `${Math.round(val * 100)}%`;
      b.style.backgroundSize = `${pct} ${pct}`;
      b.textContent = pressed ? `Button ${i} [PRESSED]` : `Button ${i}`;

>       b.style.color = pressed ? "#42f593" : "#2e2d33";

Reviewer: As we set a class name on the next line, this should be in the CSS.

Author: I didn't find a css code snippet on this mdn page, where css class defs should be placed?

Reviewer: Yes, as the CSS is not here, let's keep this that way.

      b.className = pressed ? "button pressed" : "button";
    });

    const axes = d.getElementsByClassName("axis");
    controller.axes.forEach((axis, i) => {
      const a = axes[i];
      a.textContent = `${i}: ${axis.toFixed(4)}`;
      a.setAttribute("value", axis + 1);
    });
  }

  requestAnimationFrame(updateStatus);
}

function scangamepads() {
  const gamepads = navigator.getGamepads();
  document.querySelector("#noDevices").style.display = gamepads.filter(Boolean)
    .length
    ? "none"

>     : "block";

Reviewer: What is this for?

Author: Displaying a text to the user informing no devices were found

Author: Up to now nothing is showed (empty/blank page) if no controllers are connected or no buttons were pressed

Reviewer: 👍

  for (const gamepad of gamepads) {

>     if (gamepad) {
>       // Can be null if disconnected during the session

Reviewer: ```suggestion
    // gamepad is null if disconnected during the session
    if (gamepad) {
```

window.addEventListener("gamepaddisconnected", disconnecthandler);

if (!haveEvents) {

>   document.querySelector("#noDevices").style.display = "block";

Reviewer: What is this for?

Author: Just removed, as L304 is showing a text to the user when no gamepads were found
